signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1049,6 +1052,102 @@ async fn run_dkg_from_scratch() {
     }
 }
 
+/// Tests that dkg will not be triggered if signer set didn't change
+#[test(tokio::test)]
+async fn dont_run_dkg_if_signer_set_didnt_change() {
+    let mut rng = get_rng();
+    let db = testing::storage::new_test_database().await;
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_mocked_clients()
+        .modify_settings(|settings| {
+            settings.signer.dkg_target_rounds = std::num::NonZero::<u32>::new(1).unwrap();
+        })
+        .build();
+    let config_signer_set = ctx.config().signer.bootstrap_signing_set.clone();
+
+    // Sanity check
+    assert!(!config_signer_set.is_empty());
+
+    // Write dkg shares so it won't be a reason to trigger dkg.
+    let dkg_shares = model::EncryptedDkgShares {
+        dkg_shares_status: model::DkgSharesStatus::Verified,
+        ..Faker.fake_with_rng(&mut rng)
+    };
+    db.write_encrypted_dkg_shares(&dkg_shares)
+        .await
+        .expect("failed to write dkg shares");
+
+    // We need to construct chaintip
+    let bitcoin_block: model::BitcoinBlock = Faker.fake_with_rng(&mut rng);
+    db.write_bitcoin_block(&bitcoin_block)
+        .await
+        .expect("failed to write bitcoin block");
+
+    // Check that with correct rotate keys tx we won't proceed with dkg.
+    let chaintip = db
+        .get_bitcoin_canonical_chain_tip_ref()
+        .await
+        .expect("failed to get chain tip")
+        .expect("no chain tip");
+
+    ctx.inner
+        .state()
+        .update_current_signer_set(config_signer_set.iter().cloned().collect());
+    assert!(!should_coordinate_dkg(&ctx, &chaintip).await.unwrap());
+    assert!(assert_allow_dkg_begin(&ctx, &chaintip).await.is_err());
```

Comment:
We should also have these assertions above, although it's less clear where.

Reply:
[5b34a19](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/5b34a19ef412f70b7494bff329363e1d48f82614) and one more commit add this checks